integration/test_jobs.py

```diff
@@ -277,3 +277,58 @@ def test_results_not_available_error(
     assert isinstance(execute_results[0].download_result(), BackendResult)
 
     assert isinstance(execute_results[0].download_backend_info(), BackendInfo)
+
+
+def test_submit_under_user_group(
+    _authenticated_nexus_circuit_ref: CircuitRef,
+    qa_project_name: str,
+    qa_circuit_name: str,
+) -> None:
+    """Test that a user can submit jobs under a user_group that
+    they belong to.
+
+    Requires that the test user is a member of a group called:
+    'QA_IntegrationTestGroup',
+    and not a member of a group called:
+    'made_up_group'.
+    """
+
+    my_proj = qnx.projects.get(name_like=qa_project_name)
+
+    with pytest.raises(qnx_exc.ResourceCreateFailed):
+        qnx.start_compile_job(
+            circuits=[_authenticated_nexus_circuit_ref],
+            name=f"qnexus_integration_test_compile_job_{datetime.now()}",
+            project=my_proj,
+            backend_config=qnx.AerConfig(),
+            user_group="made_up_group",
+        )
+
+    qnx.start_compile_job(
+        circuits=[_authenticated_nexus_circuit_ref],
+        name=f"qnexus_integration_test_compile_job_{datetime.now()}",
+        project=my_proj,
+        backend_config=qnx.AerConfig(),
+        user_group="QA_IntegrationTestGroup",
+    )
+
+    my_circ = qnx.circuits.get(name_like=qa_circuit_name, project=my_proj)
+
+    with pytest.raises(qnx_exc.ResourceCreateFailed):
+        qnx.start_execute_job(
+            circuits=[my_circ],
+            name=f"qnexus_integration_test_execute_job_{datetime.now()}",
+            project=my_proj,
+            backend_config=qnx.AerConfig(),
+            n_shots=[10],
+            user_group="made_up_group",
+        )
+
+    qnx.start_execute_job(
+        circuits=[my_circ],
+        name=f"qnexus_integration_test_execute_job_{datetime.now()}",
+        project=my_proj,
+        backend_config=qnx.AerConfig(),
+        n_shots=[10],
+        user_group="QA_IntegrationTestGroup",
+    )
```

Comment:
Would be nice to assert the execute job finished successfully.

Reply:
Just in general? The job will fail before it gets created in the database if the user group is incorrect, and the meter reading with the group name only gets created after the job has already completed - I'm not sure if it will test any behaviour related to the user_group functionality but would slow down the test.

Do you think it would be good to check if the job completes anyway just in general?

Reply:
Mmmm, I was thinking of it as a way of checking that we passed the `user_group` information correctly to L4. In that case, though, we'd need to use the `QuantinuumConfig` and a device like `H1-1E`.